cocotb/scheduler.py

```diff
@@ -459,7 +455,6 @@ def _event_loop(self, trigger):
                 # to try and avoid them being destroyed at a weird time (as
                 # happened in gh-957)
                 del trigger
-                del coro
```

Comment:
@eric-wieser You introduced this, is this critical?

Reply:
I think it's a bad idea to delete this line. What's the pylint error we're addressing here?

Reply:
`cocotb/scheduler.py:462:20: W0631: Using possibly undefined loop variable 'coro' (undefined-loop-variable)`

Reply:
Is there a way for coro to exist at this point? If yes, I will have learned something.

Reply:
I'm confused as to why `del` is used here. It's not guaranteed to call `__del__` immediately or ever. There is probably a better way to do it.

Reply:
@cmarqu loop variables still exist after loops finish.

Reply:
I would just mark that `noqa` for now.

Reply:
Not sure if noqa is even worth it, as pylint reports so much else...

Reply:
I've noticed... One thing that would help is configuring pylint to not report comment warnings and "not enough/too many methods/variables/attributes" warnings which I think are total bunk.

Reply:
Moving the `del` into the loop would fix the warning, and behave practically identically

Reply:
The warning is telling us that `del` could raise NameError if the loop never runs, which is true.

Reply:
> It's not guaranteed to call `__del__` immediately or ever

True, but in practice it _is_ predictable in CPython, as long as we avoid reference cycles.

Reply:
Moved the ``del coro`` inside the loop.

Reply:
> One thing that would help is configuring pylint to not report comment warnings and "not enough/too many methods/variables/attributes" warnings which I think are total bunk.

The "too-many" warnings are actually already ignored in #2181. (You may have to use the ``--rcfile`` option to make pylint use it.)